### PR TITLE
✨[CAPD] make CAPD ready for self-hosted tests

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR makes it possible to use CAPD for testing CAPI in a self-hosted cluster

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/2903

xref #2753
xref #2637
xref #2636

/assign @vincepri
/assign @sedefsavas

### CODE REVIEW

File: test/infrastructure/docker/docker/machine.go [hunk 1]
```diff
@@ -149,6 +154,50 @@ func (m *Machine) Create(ctx context.Context, role string, version *string) erro
 	return nil
 }
 
+func kindMounts(mounts []infrav1.Mount) []v1alpha4.Mount {
+	var ret []v1alpha4.Mount
```
Comment (vincepri) [inline test/infrastructure/docker/docker/machine.go:158]:
```suggestion
	ret := make([]v1alpha4.Mount, 0, len(mounts))
```

should fix the linter issue

Comment (fabriziopandini) [reply test/infrastructure/docker/docker/machine.go]: fixed!

**Comment by fabriziopandini** [comment]
/test pull-cluster-api-capd-e2e
failure seems related to a timeout, not to the current PR

File: test/infrastructure/docker/docker/machine.go [hunk 2]
```diff
@@ -149,6 +154,54 @@ func (m *Machine) Create(ctx context.Context, role string, version *string) erro
...
 
+func kindMounts(mounts []infrav1.Mount) []v1alpha4.Mount {
+	if len(mounts) == 0 {
+		return nil
+	}
+
+	ret := make([]v1alpha4.Mount, 0, len(mounts))
+	for _, m := range mounts {
+		ret = append(ret, v1alpha4.Mount{
+			ContainerPath: m.ContainerPath,
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply test/infrastructure/docker/docker/machine.go:165]: In order to do proper validation, there should be a validating webhook that now is missing in CAPD. 
But I think it is acceptable for now and we can iterate in future

**Comment by fabriziopandini** [comment]
@sedefsavas comment addressed, thanks for feedbacks!

**Pull request merged**